include/api/CGlobalIdDataCategorizer.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+#ifndef INCLUDED_ml_api_CGlobalIdDataCategorizer_h
+#define INCLUDED_ml_api_CGlobalIdDataCategorizer_h
+
+#include <model/CDataCategorizer.h>
+
+#include <api/CCategoryIdMapper.h>
+#include <api/CGlobalCategoryId.h>
+#include <api/ImportExport.h>
+
+#include <functional>
+#include <string>
+
+namespace ml {
+namespace core {
+class CStatePersistInserter;
+class CStateRestoreTraverser;
+}
+namespace model {
+class CCategoryExamplesCollector;
+}
+namespace api {
+class CJsonOutputWriter;
+
+//! \brief
+//! Wraps a categorizer, mapping local IDs to global IDs.
+//!
+//! DESCRIPTION:\n
+//! Exposes the functionality of the model library data categorizer
+//! classes, but with local category IDs mapped to the corresponding
+//! global category IDs.
+//!
+//! IMPLEMENTATION DECISIONS:\n
+//! The persistence for this class is unusual in that it is not
+//! persisted within a new sub-level in the state.  The reason for
+//! this is that the objects it contains used to be persisted by
+//! the CFieldDataCategorizer when there was only one low level
+//! categorizer.  An extra level cannot be introduced now, as that
+//! would break backwards compatibility of the state.
+//!
+class API_EXPORT CGlobalIdDataCategorizer {
```

Comment:
I know why you've call this `CGlobalIdDataCategorizer`, but to me this feels more like an implementation detail. I'd be inclined to call it something which captures the fact that it is categorising a single field, maybe `CSingleFieldDataCategorizer`?